lib/src/ast_extensions.dart

```diff
@@ -125,6 +125,25 @@ extension AstIterableExtensions on Iterable<AstNode> {
   /// but allow a split if there are elements or comments inside.
   bool canSplit(Token rightBracket) =>
       isNotEmpty || rightBracket.precedingComments != null;
+
+  /// Returns `true` if the collection containing these elements and terminated
+  /// by [rightBracket] contains any line comments before, between, or after
+  /// any elements.
+  ///
+  /// Comments within an element are ignored.
+  bool containsLineComments([Token? rightBracket]) {
+    // Look before each element.
```

Comment:
[nit] Line comments are leaning towards redundant against the code here - consider if any should be removed.